core/src/test/java/mediumtest/fixtures/SonarLintBackendFixture.java

```diff
@@ -242,7 +236,15 @@ public SonarLintBackendImpl build(FakeSonarLintClient client) {
       return sonarLintBackend;
     }
 
-    public SonarLintBackendImpl build() {
+    private static Path tempDirectory(String prefix) {
+      try {
+        return Files.createTempDirectory(prefix);
```

Comment:
Will this folder be deleted after tests are run?

Reply:
yes, it's in the shutdown method of the `SonarLintTestBackend`.

Regarding using a JUnit 5 extension, I thought about it and there are 2 reasons why I did not move forward:
* this [issue](https://github.com/junit-team/junit5/issues/1786) suggests that an extension cannot rely on `@TempDir`. We would need to manually manage those files as described in the comments
* The backend usually has different parameters in each test, so the extension would have to be a wrapper around the creation of the backend builder, which would make things a bit more complicated